### PR TITLE
rubocop: ruby 3, use new cops and take autocorrect for violations

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

set rubocop to ruby 3.0, reduce noise for next sprint

## How was this change tested?

running rubocop and circleci

### CODE REVIEW

File: app/models/complete_moab.rb [hunk 1]
```diff
@@ -29,7 +29,7 @@ class CompleteMoab < ApplicationRecord
   #  assign the other complete_moab to preserved_objects_primary_moab
   has_one :preserved_objects_primary_moab, dependent: :destroy
 
-  validates :moab_storage_root, :preserved_object, :status, :version, presence: true
```
Comment (jcoyne) [inline app/models/complete_moab.rb:32]: Are we dropping these validations intentionally?  Why?

File: app/models/preserved_object.rb [hunk 2]
```diff
@@ -68,7 +67,7 @@ def audit_moab_version_replication!
   # Of those eligible for replication, use whichever was checksum validated most recently.
   # @return [String, nil] The storage location (storage_root/storage_trunk) where the Moab that should be replicated lives.
   def moab_replication_storage_location
-    moabs_eligible_for_replication.joins(:moab_storage_root).order(last_checksum_validation: :desc).limit(1).pluck(:storage_location).first
+    moabs_eligible_for_replication.joins(:moab_storage_root).order(last_checksum_validation: :desc).limit(1).pick(:storage_location)
```
Comment (jcoyne) [inline app/models/preserved_object.rb:70]: Won't this now return an Array, where it returned a scalar before?

File: app/models/zip_part.rb [hunk 3]
```diff
@@ -16,7 +16,7 @@ class ZipPart < ApplicationRecord
     'replicated_checksum_mismatch' => 3
   }
 
-  validates :zipped_moab_version, :create_info, presence: true
+  validates :create_info, presence: true
```
Comment (jcoyne) [inline app/models/zip_part.rb:19]: Why are we dropping this validation?

File: app/models/zipped_moab_version.rb [hunk 4]
```diff
@@ -12,7 +12,7 @@ class ZippedMoabVersion < ApplicationRecord
   belongs_to :zip_endpoint, inverse_of: :zipped_moab_versions
   has_many :zip_parts, dependent: :restrict_with_exception, inverse_of: :zipped_moab_version
 
-  validates :preserved_object, :version, :zip_endpoint, presence: true
+  validates :version, presence: true
```
Comment (jcoyne) [inline app/models/zipped_moab_version.rb:15]: Why are we dropping these validations?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: app/models/preserved_object.rb (hunk 2 above)
Comment (ndushay) [reply app/models/preserved_object.rb:70]: pick takes the first:  https://til.hashrocket.com/posts/wcwerkn2il-rails-6-new-activerecord-method-pick

File: app/models/zip_part.rb (hunk 3 above)
Comment (ndushay) [reply app/models/zip_part.rb:19]: Rails/RedundantPresenceValidationOnBelongsTo

File: app/models/zipped_moab_version.rb (hunk 4 above)
Comment (ndushay) [reply app/models/zipped_moab_version.rb:15]: Rails/RedundantPresenceValidationOnBelongsTo

**Review by ndushay — COMMENTED** [review]
JCoyne:  I responded to your questions.  I blithely assumed that the rubocop rules know what they are doing, and the validation rule has the word "redundant" in it which gave me reassurance.

https://www.bigbinary.com/blog/rails-5-makes-belong-to-association-required-by-default

We don't seem to have strayed from the default configuration of "required" and these "belongs_to" statements are generally 4 years old ...

**Review by jcoyne — APPROVED** [review]
Sounds good, but those validation changes are causing the tests to fail.

File: .rubocop.yml [hunk 5]
```diff
@@ -373,7 +373,7 @@ Rails/CompactBlank: # new in 2.13
 Rails/DurationArithmetic: # new in 2.13
   Enabled: true
 Rails/RedundantPresenceValidationOnBelongsTo: # new in 2.13
-  Enabled: true
+  Enabled: false # NOTE: we want this to be false!
```
Comment (jcoyne) [inline .rubocop.yml:376]: Can you leave a note here about why we want it to be false? "Why" is always the question I wonder about when I look to comments.

Comment (ndushay) [reply .rubocop.yml]: I switched this back to true and fixed the tests.  In most cases we weren't doing explicit validation in the code, so I only had to change that in one file.

File: app/models/complete_moab.rb [hunk 6]
```diff
@@ -29,7 +31,7 @@ class CompleteMoab < ApplicationRecord
...
+  validates :status, :version, presence: true
```
Comment (ndushay) [inline app/models/complete_moab.rb:34]: this is the only spot where I changed the explicit validations in the code, and I added the comment above "belongs_to" for good measure.

**Comment by ndushay** [comment]
@jcoyne ready for one last review of yours (tests green, left the redundant presence validation rule on and fixed the tests)

**Pull request merged**